### PR TITLE
Onboarding: Update the FSE Beta flow

### LINKED ISSUE #57092: FSE beta: move opt-in to the first step
### Details

Based on the discussion in paYE8P-13J-p2, we should move the opt-in screen to the first step of the site creation flow.

### Checklist

_No response_

### Related

_No response_

### LINKED ISSUE #57093: FSE beta: polish the opt-in step design
### Details

Here is the new design proposal from @olaolusoga that we should incorporate for our FSE beta opt-in step.

screen-shot-2021-10-15-at-5 52 54-pm

### Checklist

_No response_

### Related

https://github.com/Automattic/wp-calypso/issues/57092

### PR DESCRIPTION
**Pull request by Copons**
#### Changes proposed in this Pull Request

* Move the FSE Beta opt-in step in first position for users eligible for the FSE Beta.
* Update the FSE Beta opt-in step design.

| Before | After |
| - | - |
| Screenshot 2021-10-13 at 12 49 43 | Screenshot 2021-10-19 at 17 55 04 |

#### The New FSE Beta Flow

As discussed in paYE8P-13J-p2, we should move the FSE Beta opt-in step in first position in the onboarding flow.
As far as I can see, Gutenboarding was designed to have the `IntentGathering` step always in first position, among other reasons because it's well integrated in the whole routing logic.
In a [previous attempt](https://github.com/Automattic/wp-calypso/pull/57155), I tried to conditionally render the FSE Beta opt-in step _inside_ `IntentGathering`, but eventually I realized that it would have messed with the browser's history more than I'd deem reasonable.

This approach, instead, takes advantage of the routing setup and, **for users eligible for the FSE Beta**: 

1. Redirects the root path (`/`) to the FSE Beta opt-in step (`/beta`).
2. Renders `IntentGathering` in a new `/title` path (instead of `/` which now gets redirected).
3. Adds `/beta` and `/title` as first and second steps of the FSE Beta flow.
4. Gives a back button to `IntentGathering` now that it's the second step.

#### Why Fixing Two PRs at Once?

This PR fixes both the issue for updating the FSE Beta opt-in step design, and for moving it in first position.
The reason is that the new design was created to be in first position in the first place (for example, it does not have a back button).

#### Why a New FSE Beta Hook?

Here I have created a new `useFseBetaOptInStep` hook, in addition to the existing `useFseBetaEligibility`.

The new hook determines if a user can see the FSE Beta opt-in step.
From what I could gather by poking around Gutenboarding, this happens when:
- The user is eligible for the FSE Beta (of course).
- The onboarding is not using the Anchor integration flow.

I'm not familiar with the Anchor flow, but it's a simple check (that also happens pretty much everywhere in Gutenboarding) that couldn't hurt.

Now the old `useFseBetaEligibility` hook is only ever used inside the new one.
I've decided to keep them separate nonetheless in order to have them take care of very different concerns independently.

#### Testing instructions

* Open Gutenboarding at `http://calypso.localhost:3000/new`.
* Make sure you are automatically redirected to `/new/beta`.
* It shouldn't have created a new history stack item, so using the browser's back button should return to the empty tab (or whatever you have set up).
* Test the new design, and make sure it looks good on any screen size.
* Move around Gutenboarding, both with the on-screen buttons, and the browser's ones.
* Make sure it works as expected.

To test the regular flow, we need to manually turn off the FSE Beta feature flag.
* Open `http://calypso.localhost:3000/new?flags=-full-site-editing/beta-opt-in`.
* Make sure the first step is `IntentGathering` (the title screen) and that there hasn't been any redirects (the path is still `/new`).
* Move around Gutenboarding, both with the on-screen buttons, and the browser's ones.
* Make sure it works as expected.

Fixes #57092
Fixes #57093

### CODE REVIEW

File: client/landing/gutenboarding/onboarding-block/edit.tsx [hunk 1]
```diff
@@ -200,13 +214,19 @@ const OnboardingEdit: React.FunctionComponent< BlockEditProps< Attributes > > =
 			) }
 			<Switch>
 				<Route exact path={ makePath( Step.IntentGathering ) }>
-					{ isAnchorFmPodcastIdError ? <AnchorError /> : <AcquireIntent /> }
+					{ initialStep }
```
Comment (zaguiini) [inline client/landing/gutenboarding/onboarding-block/edit.tsx]: Don't you need to call this function? Do you need it to be a function at all?

Maybe

```tsx
{canUseFseBetaOptInStep ?
  <Redirect to={ Step.FseBetaOptIn } />
: isAnchorFmPodcastIdError ?
  <AnchorError />
: <AcquireIntent />}
```

would work as well.

**Review by zaguiini — COMMENTED** [review]
I really liked the new design! Much cleaner! Left a question.

File: client/landing/gutenboarding/onboarding-block/edit.tsx (hunk 1 above)
Comment (Copons) [reply client/landing/gutenboarding/onboarding-block/edit.tsx]: > Don't you need to call this function? 

Yes, that would have been my guess too. But... it works anyway. 😅 
I wonder if it's React Router that calls it automagically?
 
> Do you need it to be a function at all?
>
> Maybe
> 
> ```tsx
> {canUseFseBetaOptInStep ?
>   <Redirect to={ Step.FseBetaOptIn } />
> : isAnchorFmPodcastIdError ?
>   <AnchorError />
> : <AcquireIntent />}
> ```
> 
> would work as well.

I could have sworn that there is a WP coding guideline forbidding ternary chains, but I can't find it, so...
It looks great, it's still readable, and it also removes that function call mystery from the code!

EDIT: oh, found it! It's the [`no-nested-ternary`](https://eslint.org/docs/rules/no-nested-ternary) eslint rule! 😄

**Comment by Copons** [comment]
In 973188e I've added a new condition that disables the FSE Beta flow during E2E tests.
I'm not particularly happy with this decision, but I can't think of a more appropriate way, considering that the FSE Beta flow appears based on: the environment (currently enabled by default on development and Horizon) and the user ID (currently enabled for users with ID ending in `00` to `09`, but this will change during the beta period).

Tests expect to start Gutenboarding in the `IntentGathering` step, and to input (or skip) the site title.
In the FSE Beta flow, Gutenboarding starts with the opt-in step instead. This will obviously fail Gutenboarding tests.

---

Also note that the current E2E (Mobile) fails, but I don't think it's related to these changes.
It's the same fail happening on @mattwiebe's PR https://github.com/Automattic/wp-calypso/pull/57066#issuecomment-948072135, and it doesn't happen for me locally (tested: both development and production, both mobile and desktop viewports).

**Review by zaguiini — APPROVED** [review]

**Review by mattwiebe — APPROVED** [review]
Thanks for looking into the e2e stuff @Copons 

This tested as expected for me, and I dig the cleaner layout.

**Review by vindl — COMMENTED** [review]
~I found the button positioning on this screen inconsistent with the rest of the flow. In all of the steps that follow, action buttons are placed in the top-right corner. Should we do the same here?~

Actually never mind, it's the same in prod now and not something that's been introduced with this PR.

**Review by vindl — APPROVED** [review]

**Pull request merged**